python-sdk/src/astro/extractors/extractor.py

```diff
@@ -0,0 +1,55 @@
+from typing import Dict, List, Optional
+
+from airflow.models.taskinstance import TaskInstance
+from openlineage.airflow.extractors import TaskMetadata
+from openlineage.airflow.extractors.base import BaseExtractor
+from openlineage.airflow.utils import get_job_name
+from openlineage.client.facet import BaseFacet
+from openlineage.client.run import Dataset as OpenlineageDataset
+
+
+class OpenLineageFacets:
+    def __init__(
+        self,
+        inputs: List[OpenlineageDataset],
+        outputs: List[OpenlineageDataset],
+        run_facets: Dict[str, BaseFacet],
+        job_facets: Dict[str, BaseFacet],
+    ):
+        self.inputs = inputs
+        self.outputs = outputs
+        self.run_facets = run_facets
+        self.job_facets = job_facets
+
+
+class PythonSDKExtractor(BaseExtractor):
+    """
+    This extractor provides visibility on what different python-sdk operator does by
+    extracting operator specific facets by calling get_openlineage_facets on each
+    operator
+    """
+
+    @classmethod
+    def get_operator_classnames(cls) -> List[str]:
+        return ["LoadFileOperator"]
```

Comment:
You're going to provide list of all extractors that will have `get_openlineage_facets` implemented right?

Still thinking if this extractor and related classes should be part of `openlineage-airflow` library.

Reply:
We will have a list of Operators with `get_openlineage_facets` implemented

Reply:
I would not like us to maintain this static list, can we not check if the operator has `get_openlineage_facets` method or not? If it has we want to extract information out of it?

Do we need a change in openlineage-airflow?

cc @julienledem @mobuchowski

Reply:
Yeah looks like if we don't want to maintain a static list we need to change the following line in OL

https://github.com/OpenLineage/OpenLineage/blob/a03989b2c1384979e493df3e36b26da2e0a68cad/integration/airflow/openlineage/airflow/extractors/extractors.py#L82-L83

Reply:
If we don't remove the need for "static list" in OL, we won't get lineage data for an inherited operator

Reply:
@kaxil I see it's exactly what I wrote here 😉 yes, I think we want "default" extractor that checks for the method in openlineage-airflow. Especially since `astronomer-providers` want to do similar stuff AFAIK.
https://github.com/astronomer/astro-sdk/pull/1025/files/53013ecbd79bd73aba37f595599cf756f1a5ea01#r993383428